gh-128384: Use a context variable for warnings.catch_warnings (#130010)
python/cpython: Doc/howto/free-threading-python.rst

expected to be reduced in upcoming Python release.   We are aiming for an
overhead of 10% or less on the pyperformance suite compared to the default
GIL-enabled build.

Behavioral changes
==================

This section describes CPython behavioural changes with the free-threaded
build.

Context variables
-----------------

> In the free-threaded build, the flag :data:`~sys.flags.thread_inherit_context`
> is set to true by default.  In the default GIL-enabled build, the flag
> defaults to false.  This will cause threads created with
> :class:`threading.Thread` to start with a copy of the
> :class:`~contextvars.Context()` of the caller of
> :meth:`~threading.Thread.start`.  If the flag is false, threads start with an
> empty :class:`~contextvars.Context()`.

Reviewer: ```suggestion
In the free-threaded build, the flag :data:`~sys.flags.thread_inherit_context`
is set to true by default which causes threads created with
:class:`threading.Thread` to start with a copy of the
:class:`~contextvars.Context()` of the caller of
:meth:`~threading.Thread.start`.  In the default GIL-enabled build, the flag
defaults to false so threads start with an
empty :class:`~contextvars.Context()`.
```
